gpii/node_modules/settingsHandlers/src/WebSocketsSettingsHandler.js

```diff
@@ -0,0 +1,154 @@
+/*!
+  GPII Web Sockets Settings Handler
+
+  Copyright 2014 Emergya
+
+  Licensed under the New BSD license. You may not use this file except in
+  compliance with this License.
+
+  You may obtain a copy of the License at
+  https://github.com/gpii/universal/LICENSE.txt
+*/
+
+/*global require*/
+
+(function () {
+
+    "use strict";
+
+    var fluid = require("infusion"),
+        gpii = fluid.registerNamespace("gpii");
+
+    fluid.registerNamespace("gpii.settingsHandlers");
+    fluid.registerNamespace("gpii.settingsHandlers.WebSocketsSettingsHandler");
+
+    // This global object keeps in memory the connected clients.
+    // Clients are added and removed respectively through addBrowser and
+    // removeBrowser functions
+    //
+    gpii.settingsHandlers.WebSocketsSettingsHandler.clients = {};
+
+    // This global object stores the current settings for every browser
+    //
+    gpii.settingsHandlers.WebSocketsSettingsHandler.settings = {};
+
+    gpii.settingsHandlers.WebSocketsSettingsHandler.getSettings = function (payload, id) {
+        var result = fluid.copy(payload);
+
+        for (var i=0; i< payload.length; i++) {
+            /*jshint -W083 */
+            fluid.each(payload[i].settings, function (v,k) {
+                result[i].settings[k] = gpii.settingsHandlers.WebSocketsSettingsHandler.settings[id].settings[k];
+            });
+        }
+        return result;
+    };
+
+    gpii.settingsHandlers.WebSocketsSettingsHandler.get = function (payload) {
+        var result = fluid.copy(payload);
+        for (var appId in payload) {
+            if (appId in gpii.settingsHandlers.WebSocketsSettingsHandler.settings) {
+                result[appId] = gpii.settingsHandlers.WebSocketsSettingsHandler.getSettings(payload[appId], appId);
+            } else {
+                result[appId] = {};
+            }
+        }
+        return result;
+    };
+
+    gpii.settingsHandlers.WebSocketsSettingsHandler.applySettingsForId = function (settings, id) {
+        var fallbackBlock = {settings: {}};
+
+        if (!(gpii.settingsHandlers.WebSocketsSettingsHandler.settings[id])) {
+            gpii.settingsHandlers.WebSocketsSettingsHandler.settings[id] = fallbackBlock;
+        }
+
+        var currentSettings = gpii.settingsHandlers.WebSocketsSettingsHandler.settings[id].settings;
+
+        var applier = fluid.makeChangeApplier(currentSettings, {
```

Comment:
This applier can go up into the "supercomponent" when it is created and so be kept around between requests